### PR TITLE
[User Interface] add method to pipe outputs to a pager

### PR DESCRIPTION
**Pull request by manuyavuz**
Do we need a CHANGELOG for it?

### CODE REVIEW

**Comment by segiddins** [comment]
No changelog, yes a test 👍🏻

**Comment by manuyavuz** [comment]
Will look for a test :+1: Do you know a way of showing need to the block in documentation? I think it should be more descriptive, but couldn't find a convenient way.

**Comment by segiddins** [comment]
`@yields` I think

**Comment by manuyavuz** [comment]
Spec added. 🚀?

File: lib/cocoapods/user_interface.rb [hunk 1]
```diff
@@ -368,6 +368,19 @@ def warn(message, actions = [], verbose_only = false)
         warnings << { :message => message, :actions => actions, :verbose_only => verbose_only }
       end
 
+      # Pipes all output inside given block to a pager.
+      #
+      # @yield Code block in which inputs to {#puts} and {#print} methods will be printed to the piper.
+      #
+      def with_pager
+        IO.popen((ENV['PAGER'] || 'less -R'), 'w') do |io|
+          Signal.trap('INT', 'IGNORE')
```
Comment (segiddins) [inline lib/cocoapods/user_interface.rb]: Do we have to do anything to reset this?

File: lib/cocoapods/user_interface.rb [hunk 2]
```diff
@@ -368,6 +368,19 @@ def warn(message, actions = [], verbose_only = false)
...
+          UI.output_io = io
+          yield
+        end
+        UI.output_io = nil
```
Comment (segiddins) [inline lib/cocoapods/user_interface.rb]: Should be in `ensure`

File: lib/cocoapods/user_interface.rb (hunk 1 above)
Comment (manuyavuz) [inline lib/cocoapods/user_interface.rb]: Yeah, good catch :+1:

File: lib/cocoapods/user_interface.rb (hunk 2 above)
Comment (manuyavuz) [inline lib/cocoapods/user_interface.rb]: Should we raise an informative in case of rescue?

Comment (segiddins) [inline lib/cocoapods/user_interface.rb]: Nah, just putting this line inside `ensure` is sufficient

**Comment by segiddins** [comment]
:+1:

**Pull request merged**